From: Doug Ledford <dledford@redhat.com>
Date: Tue, 1 Dec 2009 19:33:23 -0500
Subject: [net] cxgb3: fix port index issue
Message-id: <1259696003-21028-9-git-send-email-dledford@redhat.com>
Patchwork-id: 21590
O-Subject: [Patch RHEL5.5] [cxgb3] Fix port index issue
Bugzilla: 516948
RH-Acked-by: David S. Miller <davem@redhat.com>

Related: bz516948

Signed-off-by: Doug Ledford <dledford@redhat.com>

diff --git a/drivers/net/cxgb3/t3_hw.c b/drivers/net/cxgb3/t3_hw.c
index d84af30..0e2256a 100644
--- a/drivers/net/cxgb3/t3_hw.c
+++ b/drivers/net/cxgb3/t3_hw.c
@@ -3669,8 +3669,12 @@ static void mc7_prep(struct adapter *adapter, struct mc7 *mc7,
 
 void mac_prep(struct cmac *mac, struct adapter *adapter, int index)
 {
+	u16 devid;
+
 	mac->adapter = adapter;
-	if (!adapter->params.vpd.xauicfg[1])
+	pci_read_config_word(adapter->pdev, 0x2, &devid);
+
+	if (devid == 0x37 && !adapter->params.vpd.xauicfg[1])
 		index = 0;
 	mac->offset = (XGMAC0_1_BASE_ADDR - XGMAC0_0_BASE_ADDR) * index;
 	mac->nucast = 1;
